src/python/pants/backend/python/tasks/pytest/plugin.py

```diff
@@ -61,7 +62,7 @@ def pytest_collection_modifyitems(self, session, config, items):
     def is_conftest(itm):
       return itm.fspath and itm.fspath.basename == 'conftest.py'
     for i, item in enumerate(list(x for x in items if not is_conftest(x))):
-      if i % self._num_shards != self._shard:
+      if crc32(str(item.nodeid)) % self._num_shards != self._shard:
```

Comment:
Would be helpful to add a comment that briefly describes why we use this approach.